sycl/include/sycl/ext/oneapi/bindless_images.hpp

```diff
@@ -1156,7 +1204,7 @@ DataT fetch_image_array(const unsampled_image_handle &imageHandle
  *  @param   color The data to write
  */
 template <typename DataT, typename CoordT>
-void write_image(const unsampled_image_handle &imageHandle [[maybe_unused]],
+void write_image(unsampled_image_handle imageHandle [[maybe_unused]],
```

Comment:
this is an ABI break.

Reply:
@bader What ABI guarantees do we give for experimental extensions? 

https://intel.github.io/llvm-docs/developer/ABIPolicyGuide.html also talks about an ABI-breaking window coming up in April. Have the dates for this window already been finalized?

@Seanst98 If it was decided that we want to give ABI guarantees for experimental features, you could look into https://github.com/intel/llvm/pull/11629 for how to guard such changes between ABI breaking windows.

Reply:
From the [spec](https://github.com/intel/llvm/blob/52d984fd910ebe6059397667fdd71d386ae82e51/sycl/doc/extensions/experimental/sycl_ext_oneapi_bindless_images.asciidoc#status)

> This is an experimental extension specification, intended to provide early access to features and gather community feedback. Interfaces defined in this specification are implemented in DPC++, but they are not finalized and may change incompatibly in future versions of DPC++ without prior notice. Shipping software products should not rely on APIs defined in this specification.

So I guess you can freely change the ABI.

Reply:
> https://intel.github.io/llvm-docs/developer/ABIPolicyGuide.html also talks about an ABI-breaking window coming up in April. Have the dates for this window already been finalized?

Started today, AFAIK.

Reply:
> From the [spec](https://github.com/intel/llvm/blob/52d984fd910ebe6059397667fdd71d386ae82e51/sycl/doc/extensions/experimental/sycl_ext_oneapi_bindless_images.asciidoc#status)
> 
> > This is an experimental extension specification, intended to provide early access to features and gather community feedback. Interfaces defined in this specification are implemented in DPC++, but they are not finalized and may change incompatibly in future versions of DPC++ without prior notice. Shipping software products should not rely on APIs defined in this specification.
> 
> So I guess you can freely change the ABI.

That's right. 👍 

ABI policy guidelines do not apply to experimental features. Here is the quote from the [ABI policy guide](https://intel.github.io/llvm-docs/developer/ABIPolicyGuide.html):

> Note: Features clearly marked as experimental are considered as an exception to this guideline.

About this comment.

> > https://intel.github.io/llvm-docs/developer/ABIPolicyGuide.html also talks about an ABI-breaking window coming up in April. Have the dates for this window already been finalized?
> 
> Started today, AFAIK.

@stdale-intel, can you confirm that by updating the guide, please? As of now it's not clear if ABI breaking changes are allowed.

> ABI breaking changes window April 18 - July 11 2023 [CLOSED]
> Next ABI breaking changes window is tenatively April 2024.